packages/editor/src/utils/media-upload/media-upload.js

```diff
@@ -68,11 +68,22 @@ export function mediaUpload( {
 
 	// Allowed type specified by consumer
 	const isAllowedType = ( fileType ) => {
-		return ( allowedType === '*' ) || startsWith( fileType, `${ allowedType }/` );
+		return ( ! allowedTypes ) ||
```

Comment:
This is a bit of a long line now, would be great to make this more readable, perhaps using an early return when there are no allowed types.